src/Runner.Worker/FileCommandManager.cs

```diff
@@ -322,9 +322,21 @@ public EnvFileKeyValuePairs(IExecutionContext context, string filePath)
                     var equalsIndex = line.IndexOf("=", StringComparison.Ordinal);
                     var heredocIndex = line.IndexOf("<<", StringComparison.Ordinal);
 
-                    // Heredoc style NAME<<EOF (where EOF is typically randomly-generated Base64 and may include an '=' character)
+                    // Heredoc style NAME=<<EOF (where EOF is typically randomly-generated Base64 and may include an '=' character)
                     // see https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#multiline-strings
-                    if (heredocIndex >= 0 && (equalsIndex < 0 || heredocIndex < equalsIndex))
+                    // "key=<<EOF is also considered heredoc where the key contains an =.
+                    // Also we detect heredoc if whitespace chars exist between = and <<. (ex: "key      <<EOF")
+                    var isHeredoc = heredocIndex >= 0 &&
+                    (
+                        equalsIndex < 0 ||
+                        (
+                            heredocIndex > equalsIndex &&
+                            OnlyContainsWhiteSpaceBetweenPositions(line, equalsIndex, heredocIndex)
+                        ) ||
+                        heredocIndex < equalsIndex
```

Comment:
To go along with my other comment about flipping indices, I think we can flip these to short circuit the case where heredocIndex is before equalsIndex (meaning equals was possibly part of the delimiter).
```suggestion
                        heredocIndex < equalsIndex ||
                        (
                            heredocIndex > equalsIndex &&
                            OnlyContainsWhiteSpaceBetweenPositions(line, equalsIndex, heredocIndex)
                        )
```
```suggestion
                        (
                            heredocIndex > equalsIndex &&
                            OnlyContainsWhiteSpaceBetweenPositions(line, equalsIndex, heredocIndex)
                        ) ||
                        heredocIndex < equalsIndex
```